port_ocean/core/defaults/initialize.py

```diff
@@ -203,18 +209,60 @@ async def _initialize_defaults(
     defaults = get_port_integration_defaults(
         config_class, integration_config.resources_path
     )
-    if not defaults:
+
+    if (
+        not integration_config.create_port_resources_origin
+        and integration_config.runtime.is_saas_runtime
+    ):
+        integration_config.create_port_resources_origin = CreatePortResourcesOrigin.Port
+
+    if (
+        integration_config.create_port_resources_origin
+        == CreatePortResourcesOrigin.Port
+    ):
+        logger.info(
+            "`create_port_resources_origin` set to `Port`, verifying org feature toggle"
+        )
+        org_feature_flags = await port_client.get_organization_feature_flags()
+        if ORG_USE_PROVISIONED_DEFAULTS_FEATURE_FLAG not in org_feature_flags:
+            logger.info(
+                "Although `create_port_resources_origin` was set, it was not enabled in the organizations feature flags, disabling"
+            )
+            integration_config.create_port_resources_origin = (
+                CreatePortResourcesOrigin.Ocean
+            )
+
+    if (
+        integration_config.create_port_resources_origin
+        != CreatePortResourcesOrigin.Port
+        and not defaults
+    ):
         logger.warning("No defaults found. Skipping initialization...")
         return None
 
-    if defaults.port_app_config:
+    if (
+        defaults
+        and defaults.port_app_config
+        or integration_config.create_port_resources_origin
+        == CreatePortResourcesOrigin.Port
+    ):
         await _initialize_required_integration_settings(
-            port_client, defaults.port_app_config, integration_config
+            port_client,
+            integration_config,
+            defaults.port_app_config if defaults else None,
         )
 
     if not integration_config.initialize_port_resources:
         return
```

Comment:
I think this line should be first in the function

Reply:
Fixed!